_rules/css-orientation-lock-media-queries-b33eff.md

```diff
@@ -0,0 +1,199 @@
+---
+id: b33eff
+name: Page has no orientation lock specified using CSS media queries
+rule_type: atomic
+description: |
+  This rule checks that page content is not locked to any single display orientation using CSS media queries.
+accessibility_requirements:
+ wcag20:1.3.4: # Orientation
+  forConformance: true
+  failed: not satisfied
+  passed: further testing needed
+  inapplicable: further testing needed
+input_aspects:
+- DOM Tree
+- CSS Styling
+authors:
+- Jey Nandakumar
+- Audrey Maniez
+---
+
+## Applicability
+
+The rule applies to any element that is [visible](#visible) and has a CSS `transform` property with a [rotate](https://drafts.csswg.org/css-transforms/#funcdef-transform-rotate) transform-function, set from a CSS media query with an [orientation](https://drafts.csswg.org/mediaqueries-3/#orientation) media feature (`landscape` or `portrait`).
+
+## Expectation 1
+
+Each target element is not restricted to a single display orientation using CSS rotate transform-function, whose value is a multiple of 90 degrees but not a multiple of 180 degrees.
```

Comment:
Should "single display orientation" somehow be defined? I am not sure I get why it can be locked to 180 degrees (etc.), while not to 90 degrees (etc.).

Reply:
@annethyme I believe that given two orientations `landscape` and `portrait` as possible values, it is very straight forward to understand what single display orientation means.

As for why only multiples of 90deg, here is a scenario. Say you are looking at an application on your phone in `portrait` mode, then you flip the phone to `landscape` orientation, if there is now a lock specified, then the app will rotate by 90 and go back to only `portrait` mode. Where any value that is in multiples of 180 does not have actually lock the content. Hope that helps.

Reply:
Yeah, I question whether "is not restricted to a single display orientation" actually adds anything to the expectation. It seems to me the following would work just as well, and is more straight forward:

> None of the target elements has a CSS rotate transform-function, whose value is a multiple of 90 degrees but not a multiple of 180 degrees.